src/wasm-interpreter.h

```diff
@@ -491,6 +491,22 @@ class ExpressionRunner : public Visitor<SubType, Flow> {
   }
 };
 
+// Execute an expression in global init
+class GlobalInitRunner : public ExpressionRunner<GlobalInitRunner> {
+public:
+  Flow visitLoop(Loop* curr) { WASM_UNREACHABLE(); }
```

Comment:
This code is reached when we execute the init statement for the global, which is when we instantiate the module, which is after we fully parse and validate it. That means we can assume the expression is valid, in which case none of these unreachables would be hit. Assuming we don't have a bug in our validation ;)